libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/idp/IDPCodeGeneratorActivity.java

```diff
@@ -86,6 +88,7 @@ protected void onCreate(Bundle savedInstanceState) {
         webView.setWebViewClient(new IDPWebViewClient());
         try {
             final IDPRequestHandler idpRequestHandler = new IDPRequestHandler(spConfig, userAccount);
+            loginUrl = idpRequestHandler.getLoginUrl();
```

Comment:
Sending `loginUrl` that was used by the IDP is useful in the case when SP app didn't send any `loginUrl` to the IDP. Otherwise the SP won't know which `loginUrl` was used to authenticate.